libraries/chain/webassembly/runtimes/eos-vm-oc/gs_seg_helpers.c

```diff
@@ -64,3 +70,56 @@ void* eos_vm_oc_get_bounce_buffer_list() {
    EOSVMOC_MEMORY_PTR_cb_ptr;
    return cb_ptr->bounce_buffers;
 }
+
+uint64_t eos_vm_oc_getgs_syscall() {
+   uint64_t gs;
+   arch_prctl(ARCH_GET_GS, &gs);
+   return gs;
+}
+
+uint64_t __attribute__ ((__target__ ("fsgsbase"))) eos_vm_oc_getgs_fsgsbase() {
+   return _readgsbase_u64();
+}
+
+void eos_vm_oc_setgs_syscall(uint64_t gs) {
+   arch_prctl(ARCH_SET_GS, (unsigned long*)gs); //cast to a (unsigned long*) to match local declaration above
+}
+
+void __attribute__ ((__target__ ("fsgsbase"))) eos_vm_oc_setgs_fsgsbase(uint64_t gs) {
+   return _writegsbase_u64(gs);
+}
+
+extern char** _dl_argv;
+static int eos_vm_oc_use_fsgsbase() {
+   /* ifunc resolvers run _super_ early -- before getenv() is set up even! This is relying on the layout of _dl_argv to be
+          _dl_argv
+              ↓
+      argc, argv[0], ..., argv[argc - 1], NULL, evniron0, environ1, ..., NULL
+   */
+   const int argc = *(int*)(_dl_argv - 1);
+   char** my_environ = _dl_argv + argc + 1;
+   while(*my_environ != NULL) {
+      const char disable_str[] = "SPRING_DISABLE_FSGSBASE";
+      if(strncmp(*my_environ++, disable_str, strlen(disable_str)) == 0)
+         return 0;
+   }
```

Comment:
This is the same mess as I used in bls,
https://github.com/AntelopeIO/bls12-381/pull/4#discussion_r1279807568
Don't really know what to do here. Disabling optimizations via environment variable is nice. Seems hard to believe `_dl_argv` would ever deviate.